cmd/kubeadm/app/cmd/upgrade/node.go

```diff
@@ -105,13 +107,15 @@ func newNodeOptions() *nodeOptions {
 	return &nodeOptions{
 		kubeConfigPath: constants.GetKubeletKubeConfigPath(),
 		dryRun:         false,
+		cfgPath:        "",
 		renewCerts:     true,
 		etcdUpgrade:    true,
 	}
 }
 
 func addUpgradeNodeFlags(flagSet *flag.FlagSet, nodeOptions *nodeOptions) {
 	options.AddKubeConfigFlag(flagSet, &nodeOptions.kubeConfigPath)
+	options.AddConfigFlag(flagSet, &nodeOptions.cfgPath)
```

Comment:
Is there any reason why we are adding the `--config` flag here? I think we should keep the same semantics.

Reply:
+1, i also proposed removing --config from "apply".
let's not change this for now.

Reply:
Overwrite or creation of new kubeadm-env without the --config will only add NodeRegistration.KubeletExtraArgs defaults and not grab user configuration. As a result, user define changes are lost.

Reply:
--config for upgrades is a can of worms.
given the JoinConfiguration is not stored in the cluster, it seems better to simply stop updating the env file for both "upgrade apply" and "upgrade node".

Reply:
@neolit123 That approach works too. I will like to learn more about the can of worms introduced by --config flag. 

Please let me know if I may go ahead and remove updating the env file for both "upgrade apply" and "upgrade node".

Reply:
> Please let me know if I may go ahead and remove updating the env file for both "upgrade apply" and "upgrade node".

ideally needs more comments on https://github.com/kubernetes/kubeadm/issues/1916

>  I will like to learn more about the can of worms introduced by --config flag.

for example, miss configuration in case of custom node names can render the node inoperable if a --config is passed and this config does not contain the custom node name used for creating the node.

Reply:
Thank you for the explanation. I will wait for more comments before proceeding with changes.